cmd/dp-code-list-api/main.go

```diff
@@ -1,229 +1,85 @@
-			ID:      i.ID,
-			Code:    i.Code,
-			Label:   i.Label,
-			CodeURL: "/code-lists/" + stubs.GoodsAndServicesIndices.ID + "/codes/" + i.ID,
-		})
-	}
-
-	return r
-}()
-
-var codeListStubMap = map[string]stubs.CodeList{
-	stubs.GoodsAndServicesIndices.ID: stubs.GoodsAndServicesIndices,
-	stubs.Time.ID:                    stubs.Time,
-}
-
-var codeListsMap = map[string]codeListResponse{
-	stubs.GoodsAndServicesIndices.ID: goodsAndServicesCodeListAPIResponse,
-	stubs.Time.ID:                    timeCodeListAPIResponse,
-}
-
-var codeListCodesMap = map[string]codeListCodesResponse{
-	stubs.GoodsAndServicesIndices.ID: goodsAndServicesCodeListCodesAPIResponse,
-	stubs.Time.ID:                    timeCodeListCodesAPIResponse,
-}
-
-func codeListsHandler(w http.ResponseWriter, req *http.Request) {
-	b, err := json.Marshal(&codeListsAPIResponse)
-	if err != nil {
-		log.ErrorR(req, err, nil)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-
-	w.Header().Set("Content-Type", "application/json")
-	w.Write(b)
-}
-
-func codeListHandler(w http.ResponseWriter, req *http.Request) {
-	codeListID := mux.Vars(req)["codelist_id"]
-
-	if _, ok := codeListsMap[codeListID]; !ok {
-		log.DebugR(req, "code list not found", log.Data{"code_list_id": codeListID})
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-
-	b, err := json.Marshal(codeListsMap[codeListID])
+	mongoDatastore, err := mongostore.CreateMongoDataStore(cfg.MongoDBURL)
 	if err != nil {
-		log.ErrorR(req, err, nil)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-
-	w.Header().Set("Content-Type", "application/json")
-	w.Write(b)
-}
-
-func codesHandler(w http.ResponseWriter, req *http.Request) {
-	codeListID := mux.Vars(req)["codelist_id"]
-
-	if _, ok := codeListsMap[codeListID]; !ok {
-		log.DebugR(req, "code list not found", log.Data{"code_list_id": codeListID})
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-
-	b, err := json.Marshal(codeListCodesMap[codeListID])
-	if err != nil {
-		log.ErrorR(req, err, nil)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
-
-	w.Header().Set("Content-Type", "application/json")
-	w.Write(b)
-}
-
-func codeHandler(w http.ResponseWriter, req *http.Request) {
-	codeListID := mux.Vars(req)["codelist_id"]
-	codeID := mux.Vars(req)["code_id"]
-
-	if _, ok := codeListsMap[codeListID]; !ok {
-		log.DebugR(req, "code list not found", log.Data{"code_list_id": codeListID})
-		w.WriteHeader(http.StatusNotFound)
-		return
+		log.Error(err, nil)
+		os.Exit(1)
 	}
-
-	if _, ok := codeListStubMap[codeListID].Codes[codeID]; !ok {
-		log.DebugR(req, "code not found in list", log.Data{"code_list_id": codeListID, "code_id": codeID})
-		w.WriteHeader(http.StatusNotFound)
-		return
+	router := mux.NewRouter()
+	httpErrChannel := make(chan error)
+	_ = api.CreateCodeListAPI(router, mongoDatastore)
+	httpServer := server.New(cfg.BindAddr, router)
+	httpServer.HandleOSSignals = false
+
+	shutdown := func(httpShutdown bool) {
```

Comment:
Even though this works, it might be worth some of us sitting together to implement a common way to handle graceful shutdowns. Currently we are implementing the same thing differently; this does not mean we should move it into a single package as graceful shutdown may include closing connections to different databases and different services.